code/game/gamemodes/objective.dm

```diff
@@ -54,6 +67,53 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	if(possible_targets.len > 0)
 		target = pick(possible_targets)
 
+/**
+  *Called when you want to know if owner has escaped or not.
+  */
+/datum/objective/proc/considered_escape(datum/mind/M) //find out if the owner escaped
+	if(issilicon(M.current))
+		return FALSE
+	if(isbrain(M.current))
+		return FALSE
+	if(!M.current || M.current.stat == DEAD)
+		return FALSE
+	if(SSticker.force_ending) //This one isn't their fault, so lets just assume good faith
+		return TRUE
+	if(SSticker.mode.station_was_nuked) //If they escaped the blast somehow, let them win
+		return TRUE
+	if(SSshuttle.emergency.mode < SHUTTLE_ENDGAME)
+		return FALSE
+
+	var/turf/location = get_turf(M.current)
+	if(!location)
+		return FALSE
+
+	if(istype(location, /turf/simulated/shuttle/floor4) || istype(location, /turf/simulated/floor/mineral/plastitanium/red/brig)) // Fails traitors if they are in the shuttle brig -- Polymorph
```

Comment:
This is an interesting solution to say the least!

Reply:
unfortunately shuttles can't have multiple areas define, so floor type is used.